internal/driver/deviceclient.go

```diff
@@ -105,126 +105,186 @@ func calculateAddressLength(primaryTable string, valueType sdkModel.ValueType) u
 			var res = binary.BigEndian.Uint64(dataBytes)
 			var floatResult = math.Float64frombits(res)
-			result, err = sdkModel.NewFloat64Value(req.DeviceResourceName, resTime, floatResult)
-		case sdkModel.Int16:
+			result, err = models.NewCommandValue(req.DeviceResourceName, v2.ValueTypeFloat64, floatResult)
+			if err != nil {
+				return nil, err
+			}
+			result.Origin = resTime
+		case v2.ValueTypeInt16:
 			var res = int16(binary.BigEndian.Uint16(dataBytes))
-			result, err = sdkModel.NewFloat64Value(req.DeviceResourceName, resTime, float64(res))
-			driver.Logger.Debug(fmt.Sprintf("According to the rawType %s and the value type %s, convert integer %d to float %v ", INT16, FLOAT64, res, result.ValueToString(contract.ENotation)))
-		case sdkModel.Uint16:
+			result, err = models.NewCommandValue(req.DeviceResourceName, v2.ValueTypeFloat64, float64(res))
+			if err != nil {
+				return nil, err
+			}
+			result.Origin = resTime
+			driver.Logger.Debugf("According to the rawType %s and the value type %s, convert integer %d to float %v ", INT16, FLOAT64, res, result.ValueToString())
+		case v2.ValueTypeUint16:
 			var res = binary.BigEndian.Uint16(dataBytes)
-			result, err = sdkModel.NewFloat64Value(req.DeviceResourceName, resTime, float64(res))
-			driver.Logger.Debug(fmt.Sprintf("According to the rawType %s and the value type %s, convert integer %d to float %v ", UINT16, FLOAT64, res, result.ValueToString(contract.ENotation)))
+			result, err = models.NewCommandValue(req.DeviceResourceName, v2.ValueTypeFloat64, float64(res))
+			if err != nil {
+				return nil, err
+			}
+			result.Origin = resTime
+			driver.Logger.Debugf("According to the rawType %s and the value type %s, convert integer %d to float %v ", UINT16, FLOAT64, res, result.ValueToString())
 		}
-	case sdkModel.Bool:
+	case v2.ValueTypeBool:
 		var res = false
 		// to find the 1st bit of the dataBytes by mask it with 2^0 = 1 (00000001)
 		if (dataBytes[0] & 1) > 0 {
 			res = true
 		}
-		result, err = sdkModel.NewBoolValue(req.DeviceResourceName, resTime, res)
+		result, err = models.NewCommandValue(req.DeviceResourceName, v2.ValueTypeBool, res)
+		if err != nil {
+			return nil, err
+		}
+		result.Origin = resTime
 	default:
-		err = fmt.Errorf("return result fail, none supported value type: %v", commandInfo.ValueType)
+		return nil, fmt.Errorf("return result fail, none supported value type: %v", commandInfo.ValueType)
 	}
-	driver.Logger.Debug(fmt.Sprintf("Transfer dataBytes to CommandValue(%v) successful.", result.ValueToString(contract.ENotation)))
-	return result, err
+
+	driver.Logger.Debugf("Transfer dataBytes to CommandValue(%v) successful.", result.ValueToString())
+	return result, nil
 }
 
-func TransformCommandValueToDataBytes(commandInfo *CommandInfo, value *sdkModel.CommandValue) ([]byte, error) {
-	var byteCount = calculateByteCount(commandInfo)
+func TransformCommandValueToDataBytes(commandInfo *CommandInfo, value *models.CommandValue) ([]byte, error) {
```

Comment:
Add comment for func explaining why do we need to do this?

Reply:
@hahattan Please help to add the following comment:
TransformCommandValueToDataBytes transforms the reading value to binary data which is used to transfer data via Modbus protocol.